packages/pubsub/src/connection-pool.js

```diff
@@ -190,15 +194,19 @@ ConnectionPool.prototype.createMessage = function(connectionId, resp) {
 
   var pt = resp.message.publishTime;
   var milliseconds = parseInt(pt.nanos, 10) / 1e6;
+  var data = resp.message.data;
 
   return {
     connectionId: connectionId,
     ackId: resp.ackId,
     id: resp.message.messageId,
-    data: resp.message.data,
     attributes: resp.message.attributes,
     publishTime: new Date(parseInt(pt.seconds, 10) * 1000 + milliseconds),
     received: Date.now(),
+    // using get here to prevent user from overwriting data
+    get data() {
+      return data;
```

Comment:
fyi this makes the data field read-only, but data can still be mutated. The user can call .data.slice() for example and change the length of data, which is what we were trying to avoid.